src/python/pants/bin/local_pants_runner.py

```diff
@@ -18,12 +22,43 @@
 from pants.init.target_roots_calculator import TargetRootsCalculator
 from pants.option.options_bootstrapper import OptionsBootstrapper
 from pants.reporting.reporting import Reporting
-from pants.util.contextutil import hard_exit_handler, maybe_profiled
+from pants.util.contextutil import maybe_profiled
 
 
 logger = logging.getLogger(__name__)
 
 
+class RunTrackingExiter(Exiter):
```

Comment:
lets extract `RunTrackingExiter` changes out of `local_pants_runner` since its unrelated to this change and to keep this PR limited to `pants_daemon_runner` changes.

Reply:
The changes from this file were also reverted entirely!